upload.ts

```diff
@@ -7,58 +7,69 @@ import * as tar from "tar";
 const extractMessageRegex = /<Message>(.*?)<\/Message>/;
 
 function filter(path: string): boolean {
+  // TODO: use .dockerignore
   switch (basename(path)) {
+    case ".DS_Store":
     case ".direnv":
     case ".envrc":
     case ".git":
     case ".github":
     case ".idea":
     case ".vscode":
     case "__pycache__":
+    case "defang": // our binary
+    case "defang.exe": // our binary
     case "docker-compose.yml":
     case "docker-compose.yaml":
     case "node_modules":
       return false; // omit
   }
-  return true;
+  return true; // keep
 }
 
-export async function uploadTarball(
-  putUrl: string,
-  cwd: string
-): Promise<void> {
+export async function createTarball(cwd: string): Promise<string> {
   const tempdir = await promises.mkdtemp(join(tmpdir(), "defang-build-"));
   console.debug(`Using temporary folder ${tempdir}`);
   const temppath = join(tempdir, "context.tar.gz");
 
-  try {
-    // Using stream.pipeline() instead of .pipe() to correctly handle errors
-    await stream.pipeline(
-      tar.create({ cwd, filter, gzip: true, portable: true, strict: true }, [
-        ".",
-      ]),
-      createWriteStream(temppath)
-    );
+  // Using stream.pipeline() instead of .pipe() to correctly handle errors
+  await stream.pipeline(
+    tar.create(
+      {
+        cwd,
+        filter,
+        gzip: true,
+        mtime: 315532800, // 1980-01-01 00:00:00 GMT same as Nix
```

Comment:
```suggestion
        mtime: new Date(null), // 1970-01-01 00:00:00 GMT same as Nix
```

Reply:
Actually, that should probably be *1000 because Date wants ms.

Reply:
Yup, indeed.

Reply:
I filed https://github.com/DefinitelyTyped/DefinitelyTyped/issues/67775 to address the wrong type for `mtime`